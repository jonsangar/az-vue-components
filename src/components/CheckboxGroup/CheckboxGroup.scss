.az-checkbox-group {
  $rootClass: &;

  // margin-bottom: quarter();

  //
  // Cell State
  //
  &--cell {
    .az-checkbox {
      position: relative;
      flex-direction: row-reverse;
      justify-content: space-between;
      width: 100%;
      padding: half() half();
      overflow: hidden;
      font-size: font-size(base);
      line-height: simple();
      color: color(content);
      background-color: #ffffff;

      &--checked {
        .az-checkbox__mark {
          display: none; // Elimina la marca que arrastra del modo Checbox sin celdas agrupadas.
        }
      }

      &:not(:last-child)::after {
        position: absolute;
        right: 0;
        bottom: 0;
        left: quarter();
        pointer-events: none;
        content: ' ';
        border-bottom: 1px solid color(border);
        -webkit-transform: scaleY(0.5);
        transform: scaleY(0.5);
      }

      &:active::before {
        position: absolute;
        top: 0;
        left: 0;
        z-index: 4;
        width: 100%;
        height: 100%;
        content: '';
        background-color: rgba(0, 0, 0, 0.1) !important;
      }

      .az-checkbox__base {
        display: none;
      }

      .az-checkbox__cell {
        display: block;
      }

      .az-checkbox__label {
        flex: none;
        overflow: hidden;
        line-height: simple();
        text-overflow: ellipsis;
        white-space: nowrap;
      }
    }
  }
}
