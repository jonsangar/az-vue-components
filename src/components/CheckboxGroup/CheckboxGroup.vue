<template lang="pug">
  .az-checkbox-group(:class="classObject")
    slot
</template>

<script>
export default {
  props: {
    cell: {
      type: Boolean,
      default: false,
    },
    maxOptions: {
      required: false,
    },
  },
  computed: {
    /**
     * Devuelve un listado con las clases que dependen de la configuraci√≥n
     */
    classObject() {
      return [this.cell ? 'az-checkbox-group--cell' : ''];
    },
  },
};
</script>

<style src="./CheckboxGroup.scss" lang="scss"></style>
