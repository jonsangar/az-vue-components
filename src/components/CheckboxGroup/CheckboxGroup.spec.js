import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import CheckboxGroup from './CheckboxGroup.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('CheckboxGroup.vue', () => {
  const build = () => {
    const wrapper = shallowMount(CheckboxGroup, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente CheckboxGroup con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que tiene celda', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad celda, añade la clase celda
    await wrapper.setProps({
      cell: true,
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.cell).toStrictEqual(true);
    // Comprobamos que se ha añadido el titulo con estilos por defecto
    expect(wrapper.find('.az-checkbox-group--cell').isVisible()).toBe(true);
  });
})
