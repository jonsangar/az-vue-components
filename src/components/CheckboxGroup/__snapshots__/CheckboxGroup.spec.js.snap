// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup.vue Comprueba renderizado del componente CheckboxGroup con snapshot 1`] = `<div class="az-checkbox-group"></div>`;
